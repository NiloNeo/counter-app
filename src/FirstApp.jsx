
const newMessage = 'illika!!!';

const suma = (num1, num2) => (num1 + num2);

const FirstApp = ( { title, subTitle = 'No hay SubTitle'} ) => {

    return (
        <>
            <h1> { title } </h1>
            <p> { subTitle } </p>
        </>
    )
}


export default FirstApp
