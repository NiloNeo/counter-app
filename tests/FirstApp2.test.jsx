import { render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe("Pruebas en <FirstApp />", () => {
    const title = "Hola, soy Illika";
    const subTitle = "Soy un SubTitle";

    test('Debe de hacer Match con el Snapshot', () => {
        const { container } = render(<FirstApp title = {title} />);
        expect(container).toMatchSnapshot();
    });
    
    test('Debe de mostrar "Hola, soy Illika"', () => {        
        render(<FirstApp title = {title} />);
        expect(screen.getByText(title)).not.toBeFalsy();
    });

    test('Debe de mostrar el titulo en un h1', () => {   
        render(<FirstApp title = {title} />);
        expect(screen.getByRole("heading", {level: 1}).innerHTML).toContain(title);
    });

    test('Debe de mostrar el subTitle enviado por los props', () => {   
        render(<FirstApp title = {title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(1);
    });
    
    

})