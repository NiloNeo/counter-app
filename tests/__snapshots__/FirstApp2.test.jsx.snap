// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirstApp /> Debe de hacer Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="titleid"
  >
     
    Hola, soy Illika
     
  </h1>
  <h2>
     
    No hay SubTitle
     
  </h2>
  <p>
     
    illika
     
  </p>
</div>
`;
